torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -2538,7 +2539,38 @@ void XLANativeFunctions::_propagate_xla_data(const at::Tensor& input,
   // 1) Aid XLA's InputOutputAlias.
   auto input_tensor = bridge::GetXlaTensor(input);
   auto output_tensor = bridge::GetXlaTensor(output);
-  output_tensor->data()->alias_id = input_tensor->GetUniqueId();
+  if (input_tensor->CurrentDataHandle() != nullptr ||
```

Comment:
I guess we can always use alias_id?

Reply:
haha that's what I thought but actually no. Look at my example down below
```
    // x.tensor_id = 1, x.alias_id = 1
    x = torch.randn(5,5).to(xla_device())
    // x.tensor_id = 2, x.alias_id should be 1
    x += 1
    xm.mark_step()
    // x.tensor_id =3, x.alias_id should be 2 since input tensor id will be 2
    // for this graph
    x *= 1 of 1
    xm.mark_step()
```

if we always use `alias_id`, the `alias_id` of `x` in second would be `1`, but we need it to be 2.

Reply:
in the second execution, input tensor id is `2`, we need the alias ID to always match the input tensor ID. In other world we should not carry `alias_id` across `mark_step`.

Reply:
This is a bit tricky, even the underlying buffer is aliased, we still create a new PjrtBuffer object for `x` after the first `mark_step`. That DeviceData object(wrap about pjrtbuffer) will have `data_info ` with `tensor_id` 2, since `x`'s tensor id is 2 after the first `mark_step`.

Reply:
I see so the problem is mark_step? Will the right solution to clear out alias_id after mark_step?

Reply:
we can do that too(reset alias_id to tensor id after processed the input_output_alias info). That might make this code less confuse haha.

Reply:
That sounds like a good follow up, but feel free to skip it.